README.md

```diff
@@ -23,110 +23,131 @@ $ composer require tomphp/config-service-provider
 ```php
 <?php
 
-use League\Container\Container;
-use League\Container\ServiceProvider\AbstractServiceProvider;
-use TomPHP\ConfigServiceProvider\ConfigServiceProvider;
-
-class DatabaseConnectionProvider extends AbstractServiceProvider
-{
-    protected $provides = [
-        'database_connection',
-    ];
-    
-    public function register()
-    {
-        $this->container->share('database_connection', function () {
-            return new DatabaseConnection(
-                $this->container->get('config.db.name'),
-                $this->container->get('config.db.username'),
-                $this->container->get('config.db.password')
-            );
-        });
-    }
-}
+use League\Container\Container; // or Pimple\Container
+use TomPHP\ConfigServiceProvider\ConfigureContainer;
 
-$appConfig = [
+$config = [
     'db' => [
         'name'     => 'example_db',
         'username' => 'dbuser',
         'password' => 'dbpass',
-    ]
+    ],
+
+    'di' => [
+        'services' => [
+            'database_connection' => [
+                'class' => DatabaseConnection::class,
+                'arguments' => [
+                    'config.db.name',
+                    'config.db.username',
+                    'config.db.password',
+                ],
+            ],
+        ],
+    ],
 ];
 
 $container = new Container();
-
-$container->addServiceProvider(ConfigServiceProvider::fromConfig($appConfig));
-$container->addServiceProvider(new DatabaseConnectionProvider());
+ConfigureContainer::fromArray($container, $config);
```

Comment:
@localheinz I agree that it sounds a bit weird, it's not a naming style or programming style I'd normally go for. I just wanted to find a way to people to use the library with a bare minimum learning effort.

I like "builder" style one you've suggested, but I worry it makes it easier for users to make mistakes - (i.e. forget to call `apply()` at the end). I'm not against it but I think this should be consider.

What about a different class and method name, i.e. `Configurator::configureContainerFromArray()`?

Or we could ever remove the class all together and just have a function - would that be clearer?